app/src/main/java/com/example/activeamigo/CreateAccountActivity.java

```diff
@@ -0,0 +1,136 @@
+        return false;
+    }
+
+    // Shows the dialog boxes based on the status's
+    protected void showAlert(int messageId) {
+        AlertDialog.Builder builder = new AlertDialog.Builder(CreateAccountActivity.this);
+        builder.setMessage(messageId)
+                .setPositiveButton("OK", null);
+        AlertDialog dialog = builder.create();
+        dialog.show();
+    }
+
+    // Check if the email already exists in the database
+    protected void checkEmail(String name, String emailAddress, String password, String dbName, FirebaseFirestore fs) {
+        fs.collection(dbName)
+                .whereEqualTo("Email", emailAddress)
+                .get()
+                .addOnCompleteListener(task -> {
+                    if (task.isSuccessful()) {
+                        QuerySnapshot querySnapshot = task.getResult();
+                        if (querySnapshot != null && !querySnapshot.isEmpty()) {
+                            // Email already exists, show error message
+                            showAlert(R.string.accountCreationEmailExists);
+                        } else {
+                            // Email does not exist, proceed with adding the account
+                            addAccount(name, emailAddress, password, dbName);
+                        }
+                    } else {
+                        // Error occurred while checking for email existence
+                        showAlert(R.string.queryError);
+                    }
+                });
+    }
+
+    // Adds account to database
+    protected void addAccount(String name, String emailAddress, String password, String dbName) {
+        HashMap<String, Object> accountData = new HashMap<>();
+        accountData.put("Name", name);
+        accountData.put("Email", emailAddress);
+        accountData.put("Password", password);
+
+        // Add the account data to db
+        db.collection(dbName)
+                .document()
+                .set(accountData, SetOptions.merge())
+                // Success!
+                .addOnSuccessListener(aVoid -> {
+                    showAlert(R.string.accountCreationSuccess);
+                    clearForm(false);
+                })
+                // Failure :(
+                .addOnFailureListener(e -> showAlert(R.string.accountCreationFailed));
+    }
+
+    // Used to clear the text inputs
+    // If true will only clear the passwords
+    private void clearForm(boolean pass) {
+        EditText editTextName = findViewById(R.id.editTextName);
+        EditText editTextEmailAddress = findViewById(R.id.editTextEmailAddress);
+        EditText editTextPassword = findViewById(R.id.editTextPassword);
+        EditText editTextPasswordConfirm = findViewById(R.id.editTextPasswordConfirm);
+
+        if (!pass) {
+            editTextName.setText("");
+            editTextEmailAddress.setText("");
+        }
+        editTextPassword.setText("");
+        editTextPasswordConfirm.setText("");
+    }
+
```

Comment:
/** Back button: function is not to save edited data to firebase **/
    @Override
    public boolean onOptionsItemSelected(@NonNull MenuItem item) {
        switch (item.getItemId()) {
            case android.R.id.home:
                this.finish();

                return true;
        }
        return super.onOptionsItemSelected(item);
    }
    
    Function needed to end activity with back button

Reply:
refer to https://github.com/CSE-210/Exercise-Buddy-App/blob/PreferencePage/app/src/main/java/com/example/activeamigo/PreferenceActivity.java#L187C1-L197C6